sqlmesh/utils/pydantic.py

```diff
@@ -185,3 +192,73 @@ def wrapper(cls: t.Type, v: t.Any, values: t.Any, *args: t.Any, **kwargs: t.Any)
         return func(cls, v, values_dict, *args, **kwargs)
 
     return wrapper
+
+
+T = t.TypeVar("T")
+
+
+class PydanticTypeProxy(t.Generic[T]):
+    validate: t.Callable[[t.Any], T]
+
+    @classmethod
+    def __get_validators__(cls) -> t.Iterator[t.Callable[[t.Any], T]]:
+        yield cls.validate
+
+
+def validate_list_of_strings(v: t.Any) -> t.List[str]:
+    if isinstance(v, exp.Identifier):
+        return [v.name]
+    if isinstance(v, (exp.Tuple, exp.Array)):
+        return [e.name for e in v.expressions]
+    return [i.name if isinstance(i, exp.Identifier) else str(i) for i in v]
+
+
+def validate_string(v: t.Any) -> str:
+    if isinstance(v, exp.Expression):
+        return v.name
+    return str(v)
+
+
+def bool_validator(v: t.Any) -> bool:
+    if isinstance(v, exp.Boolean):
+        return v.this
+    if isinstance(v, exp.Expression):
+        return str_to_bool(v.name)
+    return str_to_bool(str(v or ""))
+
+
+def positive_int_validator(v: t.Any) -> int:
+    if isinstance(v, exp.Expression) and v.is_int:
+        v = int(v.name)
+    if not isinstance(v, int):
+        raise ValueError(f"Invalid num {v}. Value must be an integer value")
+    if v <= 0:
+        raise ValueError(f"Invalid num {v}. Value must be a positive integer")
+    return v
+
+
+SqlGlotListOfStrings = t.List[str]
+SqlGlotString = str
+SqlGlotBool = bool
+SqlglotPositiveInt = int
```

Comment:
@izeigerman I had to add this to work with mypy along with the type ignores. I'm not sure if there is another way around it. Basically I assign a type alias and that is how mypy views these but then Pydantic properly uses them as custom types when it does the validation. Technically keeps both happy but not very clean.

Reply:
Any chance you only need this under `t.TYPE_CHECKING`? This way it can look something like:
```
if t.TYPE_CHECKING:
    ...
elif if PYDANTIC_MAJOR_VERSION >= 2:
    ...
else:
    ...
```